### PR TITLE
Improve Agents preview page load when there are no agents

### PR DESCRIPTION
**Pull request by lucianogorza**
### Description
When there are no registered agents and the `Agents preview` page is loaded, the component with the dashboard and the list of agents is always shown first for a moment, and then the component to register an agent is displayed. This PR fixes in the case where there are no registered agents so that the component with the dashboard and the list is not shown first but directly displays the component to register an agent.
 
### Issues Resolved
https://github.com/wazuh/wazuh-dashboard-plugins/issues/6175

#### With no registered Agents: Go to Modules -> Modules directory. Click `Add agent`
No Agents - Module directory

#### With no registered Agents: Go to Agents from another page
No Agents - Agents

#### With registered Agents: Go to Agents from another page and click `Deploy new agent`
With Agents - Agents

### Test
Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| With no registered Agents: Go to Modules -> Modules directory. Click `Add agent` | :black_circle: | :black_circle: | :black_circle: |
| With no registered Agents: Go to Agents from another page | :black_circle: | :black_circle: | :black_circle: |
| With registered Agents: Go to Agents from another page and click `Deploy new agent` | :black_circle: | :black_circle: | :black_circle: |

**Details**

<details>
<summary>:black_circle: With no registered Agents: Go to Modules -> Modules directory. Click `Add agent`</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With no registered Agents: Go to Agents from another page</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With registered Agents: Go to Agents from another page and click `Deploy new agent`</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by asteriscos — CHANGES_REQUESTED** [review]
CR: :x:  Changelog entry should be placed in fixed. As it's not something new o a change in the behaviour of a feature.

Test: :heavy_check_mark: 

The agent's dashboard is not seen as it loads.

### Overview -> Register Agent

https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/2c790153-65bf-4797-a327-3a31b6e7e052

### Agents initial load

https://github.com/wazuh/wazuh-dashboard-plugins/assets/9343732/6adb56c9-844f-4993-9363-f19f170ebae1

File: plugins/main/public/controllers/agent/components/agents-preview.js [hunk 1]
```diff
@@ -219,6 +220,19 @@ export const AgentsPreview = compose(
...
       const evolutionIsReady = this.props.resultState !== 'loading';
 
+      //This condition is because the angular template and the controller have a small delay to show the register agent component when there are no agents
+      //This condition must be removed when the controller is removed
+      if (
+        !this.state.agentStatusSummary.total ||
+        this.state.agentStatusSummary.total === '-'
+      ) {
+        return (
+          <div class='md-padding md-padding-top-16'>
```
Comment (yenienserrano) [inline plugins/main/public/controllers/agent/components/agents-preview.js:230]: shouldn't it be className?

**Review by yenienserrano — COMMENTED** [review]
Test: ✅ 
CR: Comment

File: plugins/main/public/controllers/agent/components/agents-preview.js (hunk 1 above)
Comment (lucianogorza) [reply plugins/main/public/controllers/agent/components/agents-preview.js]: Done

**Review by yenienserrano — APPROVED** [review]

**Pull request merged**